staging/src/k8s.io/apimachinery/pkg/util/strategicpatch/patch.go

```diff
@@ -394,7 +574,7 @@ func diffListsOfScalars(original, modified []interface{}, diffOptions DiffOption
 		}
 	}
 
-	return addList, deletionList, nil
+	return addList, uniqifyScalars(deletionList), nil
```

Comment:
I can't find the definition in this PR but I have two issues with the name "uniqifyScalars":

1. It's misspelled-- it should be `uniquifyScalars`.
2. The naming convention I have always seen is "deduplicate". Why not call this `deduplicateScalars`?